### ISSUE TITLE
Attributes of `types.FunctionType` should be available on `lambda` functions

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#567]
### Summary

ty currently emits a false-positive diagnostic on this snippet:

```py
from typing import reveal_type

x = lambda y: y

# error[unresolved-attribute] Type `(y) -> Unknown` has no attribute `__code__`
reveal_type(x.__code__)  # revealed: Unknown
```

https://play.ty.dev/84a1b067-9670-4463-82ce-0fc025997c91

All `lambda` functions are instances of `types.FunctionType`, so all attributes available on `types.FunctionType` instances should also be available on `lambda`s.

Following https://github.com/astral-sh/ruff/pull/18242 this is easy to fix: we just need to change this line so that it calls `CallableType::function_like()` rather than `CallableType::single()` (and add some tests).

**Comment by lipefree** [comment]
I can try to resolve this issue !
